File: analyze_excel.py
import pandas as pd
import sys

def analyze(filepath):
    print(f"--- Analyzing {filepath} ---")
    try:
        xls = pd.ExcelFile(filepath)
        print(f"Sheet names: {xls.sheet_names}")
        for sheet in xls.sheet_names:
            df = pd.read_excel(filepath, sheet_name=sheet, nrows=20)
            print(f"\nSheet: {sheet}")
            print("Columns:", list(df.columns))
            print("Head:")
            print(df.head(10).to_string())
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    analyze("c:/Users/tenni/OneDrive/デスクトップ/Amazon納品梱包情報変換ツール/Excel①（Amazonの梱包グループのシート）.xlsx")
    analyze("c:/Users/tenni/OneDrive/デスクトップ/Amazon納品梱包情報変換ツール/Excel②（ラクマートの輸送箱などの情報シート）.xlsx")

File: analyze_excel.js
const xlsx = require('xlsx');

function analyzeFile(filePath) {
    console.log(`\n======================================================`);
    console.log(`--- Analyzing ${filePath} ---`);
    try {
        const workbook = xlsx.readFile(filePath);
        console.log(`Sheet names: ${workbook.SheetNames.join(', ')}`);
        
        workbook.SheetNames.forEach(sheetName => {
            console.log(`\n[ Sheet: ${sheetName} ]`);
            const worksheet = workbook.Sheets[sheetName];
            // Read as 2D array
            const data = xlsx.utils.sheet_to_json(worksheet, { header: 1, defval: null });
            console.log(`Total Rows: ${data.length}`);
            if (data.length > 0) {
                console.log("First 10 rows:");
                data.slice(0, 10).forEach((row, i) => {
                    // Filter out empty arrays or too long empty trails for display
                    console.log(`Row ${i + 1}:`, JSON.stringify(row));
                });
            }
        });
    } catch (error) {
        console.error(`Error reading ${filePath}:`, error.message);
    }
}

analyzeFile("c:/Users/tenni/OneDrive/デスクトップ/Amazon納品梱包情報変換ツール/Excel①（Amazonの梱包グループのシート）.xlsx");
analyzeFile("c:/Users/tenni/OneDrive/デスクトップ/Amazon納品梱包情報変換ツール/Excel②（ラクマートの輸送箱などの情報シート）.xlsx");
